core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -455,7 +458,16 @@ object ProducerStateManager {
 
   private def deleteSnapshotFiles(dir: File, predicate: Long => Boolean = _ => true) {
     listSnapshotFiles(dir).filter(file => predicate(offsetFromFile(file))).foreach { file =>
-      Files.deleteIfExists(file.toPath)
+      try{
+        Files.deleteIfExists(file.toPath)
+      }
+      catch {
+        case e: IOException => {
+          warn(s"In deleteSnapshotFiles(), Exception happened to deleteIfExists when deleting snapshot file ${file}, error: ${e.getMessage}. Retrying. Sleep before deleteIfExists, thread Id: ${Thread.currentThread().getId}")
+          Thread.sleep(Thread.currentThread().getId % 107)
+          Files.deleteIfExists(file.toPath)
```

Comment:
why not retry the whole block....

Reply:
I put a patch to this comment here \\wade-z240\Shared\Patch. Please check